inprocgrpc/in_process.go

```diff
@@ -325,32 +325,11 @@ func (c *Channel) NewStream(ctx context.Context, desc *grpc.StreamDesc, method s
 }
 
 func makeServerContext(ctx context.Context) context.Context {
-	// NB(jh): Not ideal to have to spin up a goroutine, but we
-	// need client->server propagation of cancellations. And we
-	// can't make the server context a child of the client context
-	// because that inadvertently leaks state, potentially
-	// contaminating it (so that server context queries will be
-	// inappropriately answered with client context values).
-	// It would be nice if context package provided a way to
-	// clear all context values, so you could create a child
-	// context that is only a child in terms of cancellation,
-	// not in terms of values.
```

Comment:
What's hard and would require support in the context package is the other way around: a context that *does* have all of the values of its parent but does *not* respect deadlines/cancelations. As described, it is super-trivial to just implement without any add'l support from the context package.

Reply:
Wouldn't either one be super trivial?

```
// Merge one context for cancellation with a different context for values.
type mergedContext struct {
	cancelParent context.Context
	valuesParent context.Context
}

func (ctx mergedContext) Deadline() (time.Time, bool) {
	return ctx.cancelParent.Deadline()
}

func (ctx mergedContext) Done() <-chan struct{} {
	return ctx.cancelParent.Done()
}

func (ctx mergedContext) Err() error {
	return ctx.cancelParent.Err()
}

func (ctx mergedContext) Value(key interface{}) interface{} {
	return ctx.valuesParent.Value(key)
}

var _ context.Context = mergedContext{}
```

Reply:
Duh. You are right. I clearly didn't think much about it.

That might be something we want to "librarify" so a handler can kick off a background task (i.e. best effort async thing, so as not to hold up a response to the client) that inherits values but will not be canceled when the main handler completes.